transport/socks5/packet_dialer.go

```diff
@@ -0,0 +1,166 @@
+// Copyright 2024 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package socks5
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/Jigsaw-Code/outline-sdk/transport"
+)
+
+type packetConn struct {
+	dstAddr net.Addr
+	pc      net.Conn
+	sc      transport.StreamConn
+}
+
+var _ net.Conn = (*packetConn)(nil)
+
+func (c *packetConn) LocalAddr() net.Addr {
+	// TODO: Is this right?
+	return c.pc.LocalAddr()
+}
+
+func (c *packetConn) RemoteAddr() net.Addr {
+	return c.dstAddr
+}
+
+func (c *packetConn) SetDeadline(t time.Time) error {
+	return c.pc.SetDeadline(t)
+}
+
+func (c *packetConn) SetReadDeadline(t time.Time) error {
+	return c.pc.SetReadDeadline(t)
+}
+
+func (c *packetConn) SetWriteDeadline(t time.Time) error {
+	return c.pc.SetWriteDeadline(t)
+}
+
+func (c *packetConn) Read(b []byte) (int, error) {
+	// TODO: read header
+	buffer := make([]byte, 65536) // Maximum size for UDP packet
+	n, err := c.pc.Read(buffer)
+	if err != nil {
+		return 0, err
+	}
+	// Minimum size of header is 10 bytes
+	if n < 10 {
+		return 0, fmt.Errorf("invalid SOCKS5 UDP packet: too short")
+	}
```

Comment:
It make me nervous that you are not immediately creating a slice with the proper boundaries. Let's work on the pkt slice instead of buffer, since buffer is bigger than the packet.
```suggestion
	}
	pkt := buffer[:n]
```